playservices/src/main/java/no/finntech/android/rx/RxPlayServices.java

```diff
@@ -0,0 +1,213 @@
+                for (Api<? extends Api.ApiOptions.NotRequiredOptions> service : services) {
+                    builder.addApi(service);
+                }
+                builder.addConnectionCallbacks(new GoogleApiClient.ConnectionCallbacks() {
+                    @Override
+                    public void onConnected(Bundle bundle) {
+                        if (!subscriber.isUnsubscribed()) {
+                            subscriber.onNext(client);
+                            // we cant trigger completed, because that'll trigger a unsubscribe which invalidates this client.
+                            // also this isn't technically "completed" as the result has an open connection until disconnected.
+                        } else {
+                            subscriber.unsubscribe();
+                        }
+                    }
+
+                    @Override
+                    public void onConnectionSuspended(int reason) {
+                        subscriber.onError(new GoogleApiConnectionSuspended(reason));
+                    }
+                });
+                builder.addOnConnectionFailedListener(new GoogleApiClient.OnConnectionFailedListener() {
+                    @Override
+                    public void onConnectionFailed(ConnectionResult connectionResult) {
+                        subscriber.onError(new GoogleApiConnectionFailed(connectionResult));
+                    }
+                });
+                client = builder.build();
+                client.connect();
+                subscriber.add(Subscriptions.create(new Action0() {
+                    @Override
+                    public void call() {
+                        unsubscribeHandler.post(new Runnable() {
+                            @Override
+                            public void run() {
+                                if (client.isConnected() || client.isConnecting()) {
+                                    client.disconnect();
+                                }
+                            }
+                        });
+                    }
+                }));
+            }
+        });
+    }
+
+
+    public static Observable<GoogleApiClient> getLocationEnabled(final Activity activity, final LocationRequest locationRequest, final RxActivityResponseDelegate.RxResponseHandler responseHandler) {
+        return getLocationEnabled(null, activity, locationRequest, responseHandler);
+    }
+
+    public static Observable<GoogleApiClient> getLocationEnabled(final Observable<Boolean> getPermissionObservable, final Activity activity, final LocationRequest locationRequest, final RxActivityResponseDelegate.RxResponseHandler responseHandler) {
+        if (getPermissionObservable == null) {
+            return getLocationEnabled(RxPermission.getPermission(activity, responseHandler, Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION), activity, locationRequest, responseHandler);
```

Comment:
Do we technically need both these permissions for every locationrequest? Could probably make this more configurable.